pilot/pkg/serviceregistry/kube/controller/ambientindex.go

```diff
@@ -146,22 +146,14 @@ func (a *AmbientIndex) Lookup(key string) []*model.AddressInfo {
 
 func (a *AmbientIndex) dropWorkloadFromService(svcAddress networkAddress, workloadUID string) {
 	wls := a.byService[svcAddress]
-	// TODO: this is inefficient, but basically we are trying to update a keyed element in a list
-	// Probably we want a Map? But the list is nice for fast lookups
-	filtered := make([]*model.WorkloadInfo, 0, len(wls))
-	for _, inc := range wls {
-		if inc.ResourceName() != workloadUID {
-			filtered = append(filtered, inc)
-		}
-	}
-	a.byService[svcAddress] = filtered
+	delete(wls, workloadUID)
```

Comment:
This is the major change

Reply:
Because we lookup very often and modify infrequently. Iterate over list is faster than map. Not sure its still a big concern, this comment has been through many refactors :-)

Reply:
Map is more efficient for mutating, we are using map quite common in places like endpointShards